### PR TITLE
refactor: moved unwanted public functions from Helper.h (#766)

### PR DESCRIPTION
**Pull request by Mehdibenhadjkhelifa**
I have moved all the functions that aren't needed to be visible via the header file helper.h to a the AlgoHelper.h file under src/common while fixing all other cpp files dependent on those function also changed the cmake file under src/ to include the AlgoHelper.h file too. This was tested to be functional using the UT provided by the project ( faker-cxx-UT) all tests passed. Also used clang-format on most of the files (forgot some) that i have changed whilst working in this refactoring as prescribed in the CONTRIBUTING.md file. Hope this helped ! Cheers !

### CODE REVIEW

File: src/common/AlgoHelper.h [hunk 1]
```diff
@@ -0,0 +1,204 @@
...
+ *
+ * @param std::set of elements.
+ *
+ * @return T a random element from the std::set.
+ *
+ * @code
+ * std::set<char> chars{'a', 'b', 'c', 'd', 'e'};
+ * faker::helper::setElement(chars) // 'd'
+ * @endcode
+ */
```
Comment (cieslarmichal) [inline src/common/AlgoHelper.h:27]: we don't need docs for internal functions, could be removed, applies to all of them in this file

**Review by cieslarmichal — CHANGES_REQUESTED** [review]

**Comment by cieslarmichal** [comment]
Looks good, only minor comment :)

**Comment by cieslarmichal** [comment]
and resolve conflicts

**Comment by Mehdibenhadjkhelifa** [comment]
Hello @cieslarmichal , i'm happy that you found my contribution useful but i wanted to know how that conflict happend? i didn't manually change the code and i made sure that i was up to date in my fork before commiting changes.I'm newish to open-source contribution so it may be something obvious that i'm missing that's why i wanted to ask. Also for the docs do you want me to make a new commit with the changes and push it or how do i follow up with your request exactly? Thanks for your time and patience !

File: src/common/AlgoHelper.h (hunk 1 above)
Comment (Mehdibenhadjkhelifa) [reply src/common/AlgoHelper.h]: Job done as i think. i edited the file and removed comments as requested

**Comment by cieslarmichal** [comment]
Very good changes 👌 so I changed that file in my changes a moment ago so it has conflict now 😁

**Review by cieslarmichal — APPROVED** [review]
🔥🔥🔥

**Pull request merged**